staging/src/k8s.io/apiserver/pkg/server/options/serving.go

```diff
@@ -244,6 +251,10 @@ func (s *SecureServingOptions) ApplyTo(config **server.SecureServingInfo) error
 		if err != nil {
 			return fmt.Errorf("failed to create listener: %v", err)
 		}
+
+		if s.ProxyProtocol {
+			s.Listener = &proxyproto.Listener{Listener: s.Listener}
+		}
```

Comment:
what behavior does this change on established connections?

Reply:
is there a reason this is only done in the `if s.Listener == nil {` branch? if there's already a listener, and s.ProxyProtocol is set, shouldn't this still be wrapped?